src/logger.interceptor.ts

```diff
@@ -24,12 +25,22 @@ export class RequestInterceptor implements NestInterceptor {
     next: CallHandler<any>,
   ): Observable<any> | Promise<Observable<any>> {
     const start = new Date();
-    const ctx = context.switchToHttp();
-    const req = ctx.getRequest<Express.Request>();
-    const res = ctx.getResponse<ServerResponse>();
+    let req: Express.Request;
+    let res: ServerResponse | undefined;
+    if (context.getType() === 'http') {
+      const ctx = context.switchToHttp();
+      req = ctx.getRequest<Express.Request>();
+      res = ctx.getResponse<ServerResponse>();
+    } else if (context.getType<GqlContextType>() === 'graphql') {
+      const ctx = GqlExecutionContext.create(context).getContext();
+      req = ctx.req;
+    } else {
+      return next.handle();
+    }
+
     const method = req.method;
-    const url = req.url;
-    const route = req.route.path;
+    const url = req.originalUrl;
+    const route = context.getType() === 'http' ? req.route.path : req.baseUrl;
```

Comment:
`context.getType()` repeats multiple times, is it better to separate the  extracting `req`/`res` from the `context` into another place so that `intercept` only takes care higher level logging tasks.

Reply:
moved extracting req / res in a function in a new commit.